src/oplog/log.js

```diff
@@ -289,14 +288,15 @@ const Log = async (identity, { logId, logHeads, access, entryStorage, headsStora
           // filter out traversed and fetched hashes
           toFetch = [...toFetch, ...next, ...refs].filter(notIndexed)
           // Function to fetch an entry and making sure it's not a duplicate (check the hash indices)
-          const fetchEntries = async (hash) => {
+          const fetchEntries = (hash) => {
             if (!traversed[hash] && !fetched[hash]) {
               fetched[hash] = true
               return get(hash)
             }
           }
           // Fetch the next/reference entries
-          const nexts = await pMap(toFetch, fetchEntries)
+          const nexts = await Promise.all(toFetch.map(p => Promise.resolve(p).then(fetchEntries)))
```

Comment:
Can this be `const next = await Promise.all(toFetch.map(fetchEntries))`?

Reply:
Quite right, I thought `toFetch` might contain Promises but it doesn't. Updated.